static/js/dashboard-spa.js

```diff
@@ -0,0 +1,538 @@
+    mergeAliases(currentAliases, newAliases) {
+      // dict of aliasId and alias to speed up research
+      let newAliasesDict = {};
+      for (let i = 0; i < newAliases.length; i++) {
+        let alias = newAliases[i];
+        newAliasesDict[alias.id] = alias;
+      }
+
+      let ret = [];
+
+      // keep track of added aliases
+      let alreadyAddedId = {};
+      for (let i = 0; i < currentAliases.length; i++) {
+        let alias = currentAliases[i];
+        if (newAliasesDict[alias.id]) ret.push(newAliasesDict[alias.id]);
+        else ret.push(alias);
+
+        alreadyAddedId[alias.id] = true;
+      }
+
+      for (let i = 0; i < newAliases.length; i++) {
+        let alias = newAliases[i];
+        if (!alreadyAddedId[alias.id]) {
+          ret.push(alias);
+        }
+      }
+
+      return ret;
+    },
+
+    async loadMoreAliases() {
+      this.isLoadingMoreAliases = true;
+      this.page++;
+
+      // we already fetched aliases of the next page, just merge it
+      this.aliasesArray = this.mergeAliases(this.aliasesArray, this.aliasesArrayOfNextPage);
+
+      // fetch next page in advance to know if there is a next page
+      this.aliasesArrayOfNextPage = await this.fetchAlias(this.page + 1, this.searchString);
+
+      // use jquery multiple select plugin after Vue has rendered the aliases in the DOM
+      this.$nextTick(() => {
+        $('.mailbox-select').multipleSelect();
+        $('.mailbox-select').removeClass('mailbox-select');
+      });
+
+      this.isLoadingMoreAliases = false;
+    },
+
+    resetFilter() {
+      this.searchString = "";
+      this.filter = "";
+      this.loadAliases();
+    },
+
+    // enable or disable the 'Create' button depending on whether the alias prefix is valid or not
+    handleAliasPrefixInput() {
+      this.aliasPrefixInput = this.aliasPrefixInput.toLowerCase();
+      if (this.aliasPrefixInput.match(ALIAS_PREFIX_REGEX)) {
+        document.querySelector('.bootbox-accept').classList.remove('disabled');
+        this.aliasPrefixError = "";
+      } else {
+        document.querySelector('.bootbox-accept').classList.add('disabled');
+        this.aliasPrefixError = this.aliasPrefixInput.length > 0 ? "Only lowercase letters, numbers, dots (.), dashes (-) and underscores (_) are supported." : "";
+      }
+    },
+
+    async createCustomAlias() {
+      try {
+        const res = await fetch("/api/v3/alias/custom/new", {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            alias_prefix: this.aliasPrefixInput,
+            signed_suffix: this.aliasSelectedSignedSuffix,
+            mailbox_ids: [this.defaultMailboxId],
+            note: this.aliasNoteInput,
+          }),
+        });
+
+        if (res.ok) {
+          const alias = await res.json();
+          this.aliasesArray.unshift(alias);
```

Comment:
we can also scroll to this new alias as user can be way below the viewport by using `scrollIntoView()`